### PR TITLE
Adjust Series specific tests for string option

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.1.2.rst [hunk 1]
```diff
@@ -39,7 +39,7 @@ Bug fixes
 Other
 ~~~~~
 - Fixed non-working installation of optional dependency group ``output_formatting``. Replacing underscore ``_`` with a dash ``-`` fixes broken dependency resolution. A correct way to use now is ``pip install pandas[output-formatting]``.
--
+- Setting the environment variable ``PANDAS_INFER_STRING`` to ``"1"`` will now enable the option ``pd.options.future.infer_string`` (:issue:`55533`)^
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.1.2.rst:42]: wasnt this a separate PR?

File: pandas/_testing/asserters.py [hunk 2]
```diff
@@ -440,7 +440,10 @@ def assert_is_sorted(seq) -> None:
     if isinstance(seq, (Index, Series)):
         seq = seq.values
     # sorting does not change precisions
-    assert_numpy_array_equal(seq, np.sort(np.array(seq)))
+    if isinstance(seq, np.ndarray):
+        assert_numpy_array_equal(seq, np.sort(np.array(seq)))
```
Comment (jbrockmendel) [inline pandas/_testing/asserters.py:444]: might be faster to use libalgos.is_monotonic?

File: pandas/conftest.py [hunk 3]
```diff
@@ -1999,6 +1999,14 @@ def using_copy_on_write() -> bool:
     )
 
 
+@pytest.fixture
+def using_infer_string() -> bool:
+    """
+    Fixture to check if Copy-on-Write is enabled.
```
Comment (jbrockmendel) [inline pandas/conftest.py:2005]: docstring looks copy/pasted, needs updating

File: pandas/tests/series/indexing/test_getitem.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ def test_getitem_unrecognized_scalar(self):
     def test_getitem_negative_out_of_bounds(self):
         ser = Series(["a"] * 10, index=["a"] * 10)
 
-        msg = "index -11 is out of bounds for axis 0 with size 10"
+        msg = "index -11 is out of bounds for axis 0 with size 10|index out of bounds"
```
Comment (jbrockmendel) [inline pandas/tests/series/indexing/test_getitem.py:74]: can you use a "|".join pattern

File: doc/source/whatsnew/v2.1.2.rst (hunk 1 above)
Comment (phofl) [reply doc/source/whatsnew/v2.1.2.rst:42]: Yeah good point, removed

File: pandas/_testing/asserters.py (hunk 2 above)
Comment (phofl) [reply pandas/_testing/asserters.py:444]: This is used in one place, tbh I want to get rid of the method completely, but that wasn't in scope here

File: pandas/conftest.py (hunk 3 above)
Comment (phofl) [reply pandas/conftest.py]: Thx for catching

File: pandas/tests/series/indexing/test_getitem.py (hunk 4 above)
Comment (phofl) [reply pandas/tests/series/indexing/test_getitem.py:74]: I don't feel super strong here but I prefer the pipe pattern for 2 options

File: pandas/tests/series/accessors/test_dt_accessor.py [hunk 5]
```diff
@@ -582,13 +582,15 @@ def test_strftime_dt64_days(self):
...
-    def test_strftime_period_days(self):
+    def test_strftime_period_days(self, using_infer_string):
         period_index = period_range("20150301", periods=5)
         result = period_index.strftime("%Y/%m/%d")
         expected = Index(
             ["2015/03/01", "2015/03/02", "2015/03/03", "2015/03/04", "2015/03/05"],
             dtype="=U10",
         )
+        if using_infer_string:
+            expected = expected.astype("string[pyarrow_numpy]")
```
Comment (mroeschke) [inline pandas/tests/series/accessors/test_dt_accessor.py:593]: I forgot if/when this was discussed, but I thought `pd.ArrowDtype(pa.string())` was going to be the new default?

Comment (phofl) [reply pandas/tests/series/accessors/test_dt_accessor.py:593]: No we discussed this at the summit that we would use a string dtype that follows numpy semantics with arrow backing

Comment (jorisvandenbossche) [reply pandas/tests/series/accessors/test_dt_accessor.py:593]: @mroeschke https://github.com/pandas-dev/pandas/issues/54792 tracks that

File: doc/source/whatsnew/v2.1.2.rst [hunk 6]
```diff
@@ -60,7 +60,6 @@ Bug fixes
 Other
 ~~~~~
 - Fixed non-working installation of optional dependency group ``output_formatting``. Replacing underscore ``_`` with a dash ``-`` fixes broken dependency resolution. A correct way to use now is ``pip install pandas[output-formatting]``.
--
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.2.rst:63]: Could you undo this?

Comment (phofl) [reply doc/source/whatsnew/v2.1.2.rst]: done

File: pandas/tests/series/indexing/test_setitem.py [hunk 7]
```diff
@@ -527,14 +529,17 @@ def test_setitem_empty_series_timestamp_preserves_dtype(self):
...
+    def test_append_timedelta_does_not_cast(self, td, using_infer_string):
         # GH#22717 inserting a Timedelta should _not_ cast to int64
         expected = Series(["x", td], index=[0, "td"], dtype=object)
 
         ser = Series(["x"])
         ser["td"] = td
         tm.assert_series_equal(ser, expected)
-        assert isinstance(ser["td"], Timedelta)
+        if using_infer_string and not isinstance(td, Timedelta):
+            assert not isinstance(ser["td"], Timedelta)
```
Comment (mroeschke) [inline pandas/tests/series/indexing/test_setitem.py:540]: What would `ser["td"]` be here?

Comment (phofl) [reply pandas/tests/series/indexing/test_setitem.py:540]: one of those 2:

```
            Timedelta("9 days").to_timedelta64(),
            Timedelta("9 days").to_pytimedelta(),
```

maybe we should xfail that test, not sure anymore what my reasoning here was

File: pandas/tests/series/test_arithmetic.py [hunk 8]
```diff
@@ -491,14 +491,27 @@ def test_ser_cmp_result_names(self, names, comparison_op):
...
         exp = Series([False, False, False])
-        tm.assert_series_equal(s == s2, exp)
-        tm.assert_series_equal(s2 == s, exp)
+        if using_infer_string:
+            import pyarrow as pa
+
+            msg = "has no kernel"
+            # with tm.assert_produces_warning(DeprecationWarning, match="comparison"):
+            with pytest.raises(pa.lib.ArrowNotImplementedError, match=msg):
+                s == s2
```
Comment (mroeschke) [inline pandas/tests/series/test_arithmetic.py:506]: This should probably return all `False` in the future?

Comment (phofl) [reply pandas/tests/series/test_arithmetic.py:506]: Yes if we want to patch the arrow behaviour, makes probably sense

Comment (mroeschke) [reply pandas/tests/series/test_arithmetic.py:505]: OK could you add a TODO here as a reminder? Would be good to include it in an issue referencing things that would need fixing before 3.0

File: pandas/tests/series/indexing/test_setitem.py (hunk 7 above)
Comment (mroeschke) [reply pandas/tests/series/indexing/test_setitem.py:540]: Yeah I find it weird to begin with that this always promoted timedeltas to `pd.Timedelta`.

Agree to xfail this test for now to nail down the correct behavior in a follow up

**Comment by phofl** [comment]
I'll open issues for them and link them here

**Comment by phofl** [comment]
Opened issues for both

**Review by mroeschke — APPROVED** [review]
LGTM. Will can merge in a few days unless others have comments

**Pull request merged**